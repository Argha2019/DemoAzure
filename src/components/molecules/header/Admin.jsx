import React from 'react';
import { Navbar } from "react-bootstrap";
import { NavDropdown } from "react-bootstrap";
import { Row,Col } from "react-bootstrap";
import '../../../css/global.css';
import { withRouter } from 'react-router-dom';
import pic from '../../../OMSFinal.png';
import search from '../../../images.png';
import axios from 'axios';
import Cookies from 'js-cookie';
import { encryptByAES, decryptByAES } from '../../pages/Encryption.jsx';
import Config from "../../../config/urls.jsx";


/**
 * @author kasjain(Kashish Jain)
 * This file is responsible for making the header navigation bar for ADMIN at molecule level.
 */

type Props = {
	children: React.Node,
	className: string,
}
/**
 * This function is called when the user clicks logout in navigation bar.
 */
function logout() {
	var url = Config.serverUrl + '/signin/logout';

		axios.post(url, encryptByAES(JSON.parse(decryptByAES(Cookies.get('adminData')))[0].accessToken), {headers: {'Content-Type': 'application/json'}})
		.then(res=>

		Cookies.remove('adminData'));
}


const Header = (props: Props): React.Element<*> => {

	return (
	/**
	* This gives the navigation bar.
	* @return navigation bar for ADMIN.
	*/
		<Navbar bg="dark" expand="lg" variant="dark" fixed="top">
		<Navbar.Brand href="#/loginAdmin">
			<img src={pic} width="40px" className = "headerImage" alt="Company Logo" />
			<span className= "font">OMS</span>
		</Navbar.Brand>
		<Navbar.Collapse id="basic-navbar-nav">
		</Navbar.Collapse>
			<Row className="rowAlign" md={12}>
				<Col className="nav" md={12}>
					<Navbar.Brand href="#/loginAdmin">
						<img src={search} className = "searchHeader" alt="Search button" />
					</Navbar.Brand>
					<div className = "font">Hi {JSON.parse(decryptByAES(Cookies.get('adminData')))[0].firstName}</div>
					<NavDropdown title="⚙️" id="basic-nav-dropdown">
						<i><NavDropdown.Item className="fa fa-user" href="#/adminProfile"> <span>My Profile</span></NavDropdown.Item></i>
						<i><NavDropdown.Item className="fa fa-plus-square" href="#/register" ><span>Register User</span></NavDropdown.Item></i>
						<i><NavDropdown.Item className="fa fa-key" href="#/changePassword" > <span>Change Password</span></NavDropdown.Item></i>
						<i><NavDropdown.Item className="fa fa-sign-out" href="#/" onClick = {() =>logout()} > <span>Log out</span></NavDropdown.Item></i>
					</NavDropdown>
				</Col>
			</Row>
	</Navbar>
	);
}	
Header.defaultProps = {
	className: '',
  }
export default withRouter(Header);
