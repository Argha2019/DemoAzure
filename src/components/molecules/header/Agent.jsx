import React from 'react';
import { Navbar } from "react-bootstrap";
import { NavDropdown } from "react-bootstrap";
import { Form } from "react-bootstrap";
import { Button } from "react-bootstrap";
import '../../../css/global.css';
import { withRouter } from 'react-router-dom';
import pic from '../../../OMSFinal.png';
import search from '../../../images.png';
import { Row,Col } from "react-bootstrap";
import axios from 'axios';
import Cookies from 'js-cookie';
import { encryptByAES, decryptByAES } from '../../pages/Encryption.jsx';
import Config from "../../../config/urls.jsx";


/**
 * @author kasjain(Kashish Jain)
 * This file is responsible for making the header navigation bar for AGENT at molecule level.
 */

type Props = {
	children: React.Node,
	className: string,
}

/**
 * This function is called when the user clicks logout in navigation bar.
 */

function logout() {
	var url = Config.serverUrl + '/signin/logout';

	axios.post(url , encryptByAES(JSON.parse(decryptByAES(Cookies.get('agentData')))[0].accessToken), {headers: {'Content-Type': 'application/json'}})
		.then(res=>
			
	Cookies.remove('agentData'));
}

const Header = (props: Props): React.Element<*> => {

	return (
	/**
	* This gives the navigation bar.
	* @return navigation bar for AGENT.
	*/
		<div>
		<Navbar className="headerfont" bg="dark" expand="lg" variant="dark" fixed="top">
			<Navbar.Brand href="#/loginAgent"> 
				<img src={pic} width="40px" className = "headerImage" alt="Comapny Logo" />
				<span className= "font">OMS</span>
			</Navbar.Brand>
			<Navbar.Collapse id="basic-navbar-nav">
			</Navbar.Collapse>
			<Row className="rowAlign" md={12}>
				<Col className="nav" md={12}>
				<Navbar.Brand href="#/loginAgent">
					<img src={search} className = "searchHeader" alt="Search Button" />
				</Navbar.Brand>
				<div className = "font">Hi {JSON.parse(decryptByAES(Cookies.get('agentData')))[0].firstName}</div>
				<NavDropdown title="⚙️" id="basic-nav-dropdown">
					<i><NavDropdown.Item className="fa fa-user" href="#/agentProfile"> <span>My Profile</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-plus-square" href="#/createOrder"  ><span>Create Order</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-eye" href="#/viewOrder" > <span>View Order</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-user-plus" href="#/register" ><span>Register User</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-th" href="#/adjustInventory" > <span>Adjust Inventory</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-key" href="#/changePassword" > <span>Change Password</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-sign-out" href="#/" onClick = {() =>logout()}> <span>Log out</span></NavDropdown.Item></i>
				</NavDropdown>
				</Col>
			</Row>
		</Navbar>
		</div>
	)
}

Header.defaultProps = {
	className: '',
}
export default withRouter(Header);
