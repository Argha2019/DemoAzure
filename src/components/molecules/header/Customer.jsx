import React from 'react';
import { Navbar } from "react-bootstrap";
import { NavDropdown } from "react-bootstrap";
import { Button } from "react-bootstrap";
import '../../../css/global.css';
import pic from '../../../OMSFinal.png';
import search from '../../../images.png';
import { Row,Col } from "react-bootstrap";
import axios from 'axios';
import Cookies from 'js-cookie';
import { encryptByAES, decryptByAES } from '../../pages/Encryption.jsx';
import Config from "../../../config/urls.jsx";

/**
 * @author kasjain(Kashish Jain)
 * This file is responsible for making the header navigation bar for CUSTOMER at molecule level.
 */

type Props = {
	children: React.Node,
	className: string,
}

/**
 * This function is called when the user clicks logout in navigation bar.
 */
function logout() {
	var url = Config.serverUrl + '/signin/logout';

	axios.post(url , encryptByAES(JSON.parse(decryptByAES(Cookies.get('customerData')))[0].accessToken), {headers: {'Content-Type': 'application/json'}})
	.then(res=>
	Cookies.remove('customerData'));

}


const Header = (props: Props): React.Element<*> => {

	return (
	/**
	* This gives the navigation bar.
	* @return navigation bar for CUSTOMER.
	*/
		<div>
		<Navbar className="headerfont" bg="dark" expand="lg" variant="dark" fixed="top">
			<Navbar.Brand href="#/loginCustomer">
				<img alt="Company Logo" src={pic} width="40px" className = "headerImage"/>
				<span className= "font">OMS</span>
			</Navbar.Brand>
			<Navbar.Collapse id="basic-navbar-nav">
			</Navbar.Collapse>
			<Row className="rowAlign" md={12}>
				<Col className="nav" md={12}>
				<div className = "font">Hi {JSON.parse(decryptByAES(Cookies.get('customerData')))[0].firstName} </div>
				<NavDropdown title="⚙️" id="basic-nav-dropdown">
					<i><NavDropdown.Item className="fa fa-user" href="#/customerProfile"  > <span>My Profile</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-plus-square" href="#/createOrderCustomer"  ><span>Create Order</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-eye" href="#/viewOrderCustomer" > <span>View Order</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-key" href="#/changePassword" > <span>Change Password</span></NavDropdown.Item></i>
					<i><NavDropdown.Item className="fa fa-sign-out" href="#/" onClick = {() =>logout()} > <span>Log out</span></NavDropdown.Item></i>
				</NavDropdown>
				</Col>
			</Row>
		</Navbar>
		</div>
	)
}

Header.defaultProps = {
	className: '',
  }
export default Header;
