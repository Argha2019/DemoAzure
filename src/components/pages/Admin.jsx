import React from 'react';
import AdminTemplate from "../templates/AdminTemplate/Template.jsx";
import AgentListTemplate from "../templates/AgentListTemplate/AgentListTemplate.jsx";
import CustomerListTemplate from "../templates/CustomerListTemplate/CustomerListTemplate.jsx";
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import "react-tabs/style/react-tabs.css";

export default class Admin extends React.Component {
    constructor(props) {
        super(props)
    }
    render() {
        return (
            <AdminTemplate>
                <Tabs className="tabs">
                    <TabList>
                        <Tab>Agent Search</Tab>
                        <Tab>Customer Search</Tab>
                    </TabList>

                    <TabPanel>
                        <AgentListTemplate/>
                    </TabPanel>
                    <TabPanel>
                        <CustomerListTemplate/>
                    </TabPanel>
                </Tabs>
            </AdminTemplate>
        );
    }
}
