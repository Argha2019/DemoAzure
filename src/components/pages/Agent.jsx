import React from 'react';
import AgentTemplate from "../templates/AgentTemplate/Template.jsx";
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import AgentListTemplate from "../templates/AgentListTemplate/AgentListTemplate.jsx";
import CustomerListTemplate from "../templates/CustomerListTemplate/CustomerListTemplate.jsx";
import "react-tabs/style/react-tabs.css";

export default class Agent extends React.Component {
    constructor(props) {
        super(props)
    }

    render() {
        return (
            <AgentTemplate>
                <Tabs className="tabs">
                    <TabList>
                        <Tab>Agent Search</Tab>
                        <Tab>Customer Search</Tab>
                    </TabList>
						
                    <TabPanel>
                        <AgentListTemplate/>
                    </TabPanel>
                    <TabPanel>
                        <CustomerListTemplate/>
                    </TabPanel>
                </Tabs>
            </AgentTemplate>
        );
    }
}
