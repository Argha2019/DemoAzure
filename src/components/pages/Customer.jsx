import React from 'react';
import CustomerTemplate from "../templates/CustomerTemplate/Template.jsx";
import CustomerProfile from "./CustomerProfile.jsx";

export default class Customer extends React.Component {
    render(){
        return (
            <CustomerTemplate>
                <CustomerProfile />
            </CustomerTemplate>
        );
    }
} 