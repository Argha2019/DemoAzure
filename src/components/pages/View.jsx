import React from 'react';
import ViewTemplate from "../templates/ViewTemplate/Template.jsx";
import '../../css/global.css';
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-blue.css';
import ViewProfile from "./ViewProfile.jsx";


/**
 * @author Arghadeep Mandal
 * This file is responsible for making the header and footer template for role:"VIEW"
 * which is always rendered in every page. 
 */


export default class View extends React.Component {

    render() {
        return (
            <ViewTemplate>
                <ViewProfile />
            </ViewTemplate>
        );
    }
}
