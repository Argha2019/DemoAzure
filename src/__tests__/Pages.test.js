import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Agent from '../components/pages/Agent.jsx';
import Admin from '../components/pages/Admin.jsx';
import Customer from '../components/pages/Customer.jsx';
import View from '../components/pages/View.jsx';


//test suite for Agent Search
describe('Agent Admin Component', () => {

  //generate snapshot
   it('should render snapshot of Agent page', () => {
      const wrapper = shallow(<Agent />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

	//generate snapshot
   it('should render snapshot of Admin page', () => {
      const wrapper = shallow(<Admin />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

		//generate snapshot
   it('should render snapshot of Customer page', () => {
      const wrapper = shallow(<Customer />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

			//generate snapshot
   it('should render snapshot of View page', () => {
      const wrapper = shallow(<View />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

     
});
