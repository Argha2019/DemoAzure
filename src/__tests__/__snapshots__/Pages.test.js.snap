// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agent Admin Component should render snapshot of Admin page 1`] = `
<AdminTemplate>
  <Tabs
    className="tabs"
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <TabList
      className="react-tabs__tab-list"
    >
      <Tab
        className="react-tabs__tab"
        disabledClassName="react-tabs__tab--disabled"
        focus={false}
        id={null}
        panelId={null}
        selected={false}
        selectedClassName="react-tabs__tab--selected"
      >
        Agent Search
      </Tab>
      <Tab
        className="react-tabs__tab"
        disabledClassName="react-tabs__tab--disabled"
        focus={false}
        id={null}
        panelId={null}
        selected={false}
        selectedClassName="react-tabs__tab--selected"
      >
        Customer Search
      </Tab>
    </TabList>
    <TabPanel
      className="react-tabs__tab-panel"
      forceRender={false}
      selectedClassName="react-tabs__tab-panel--selected"
    >
      <withRouter(AgentListTemplate) />
    </TabPanel>
    <TabPanel
      className="react-tabs__tab-panel"
      forceRender={false}
      selectedClassName="react-tabs__tab-panel--selected"
    >
      <CustomerListTemplate />
    </TabPanel>
  </Tabs>
</AdminTemplate>
`;

exports[`Agent Admin Component should render snapshot of Agent page 1`] = `
<AgentTemplate>
  <Tabs
    className="tabs"
    defaultFocus={false}
    defaultIndex={null}
    forceRenderTabPanel={false}
    selectedIndex={null}
  >
    <TabList
      className="react-tabs__tab-list"
    >
      <Tab
        className="react-tabs__tab"
        disabledClassName="react-tabs__tab--disabled"
        focus={false}
        id={null}
        panelId={null}
        selected={false}
        selectedClassName="react-tabs__tab--selected"
      >
        Agent Search
      </Tab>
      <Tab
        className="react-tabs__tab"
        disabledClassName="react-tabs__tab--disabled"
        focus={false}
        id={null}
        panelId={null}
        selected={false}
        selectedClassName="react-tabs__tab--selected"
      >
        Customer Search
      </Tab>
    </TabList>
    <TabPanel
      className="react-tabs__tab-panel"
      forceRender={false}
      selectedClassName="react-tabs__tab-panel--selected"
    >
      <withRouter(AgentListTemplate) />
    </TabPanel>
    <TabPanel
      className="react-tabs__tab-panel"
      forceRender={false}
      selectedClassName="react-tabs__tab-panel--selected"
    >
      <CustomerListTemplate />
    </TabPanel>
  </Tabs>
</AgentTemplate>
`;

exports[`Agent Admin Component should render snapshot of Customer page 1`] = `
<CustomerTemplate>
  <CustomerProfile />
</CustomerTemplate>
`;

exports[`Agent Admin Component should render snapshot of View page 1`] = `
<ViewTemplate>
  <ViewProfile />
</ViewTemplate>
`;
